### PR TITLE
Expose livenessProbe and readinessProbe headers to consumer (in stable/wordpress)

### PR DESCRIPTION
**Pull request by JaredReisinger**
#### What this PR does / why we need it:

*Problem:* Despite docs that say otherwise, Kubernetes treats a 302 HTTP response as "failed" in liveness and readiness probes.  (See https://github.com/kubernetes/kubernetes/issues/47893.)  When using ingress-terminated HTTPS, it is common for HTTP requests to WordPress to result in 302-redirects to the HTTPS equivalent.  As a result, the container is considered unhealthy/unready.

*Solution:* Expose the `httpHeaders` property of the `httpGet` in the probe definition so that consumers can, for example, pass the `X-Forwarded-Proto: https` header to prevent the 302 and instead get a 200 response status.  The headers are exposed as optional `livenessProbeHeaders` and `readinessProbeHeaders` arrays, and the `httpHeaders` array is only included in the deployment probe definitions when the respective probe headers are defined.

Signed-off-by: Jared Reisinger <jaredreisinger@hotmail.com>

#### Which issue this PR fixes

_(no existing issues)_

#### Special notes for your reviewer:

* Update `deployment.yaml` to include optional headers.
* Update `values.yaml` and `values-production.yaml` with commented-out examples.
* Update `README.md` with configuration options, and a more detailed explanation in the Ingress section.
* Update `Chart.yaml` version number from `3.3.0` to `3.3.1`.

#### Checklist
[Place an '[x]' (no spaces) in all applicable fields. Please remove unrelated fields.]
- [X] [DCO](https://www.helm.sh/blog/helm-dco/index.html) signed
- [X] Chart Version bumped
- [X] Variables are documented in the README.md

### CODE REVIEW

**Comment by tompizmor** [comment]
LGTM @JaredReisinger,

Can you fix the conflicts in the `stable/wordpress/Chart.yaml` file so this can be merged?

**Comment by JaredReisinger** [comment]
Sorry for all the noise... I managed to push two commits without the `--signoff`, and it took a bit for me to unroll the madness.  It's all good now!

**Comment by JaredReisinger** [comment]
Looks like there was an error attaching to an already-in-use volume.  Let's see if this will work (might need to be the only thing in the comment?)

/ retest

**Comment by JaredReisinger** [comment]
Well... that didn't help.  The first run failed due to a "volume multi attach" problem, and the retest may have run into the chart version having already been updated once.  :-(  Any suggestion?

**Comment by tompizmor** [comment]
@JaredReisinger you need to bump the chart version again. In this PR you changed it to 5.0.3 but that version is already committed in master because of a different PR has been merged. 
You need to change it to 5.0.4

File: stable/wordpress/Chart.yaml [hunk 1]
```diff
@@ -1,5 +1,5 @@
 name: wordpress
-version: 5.0.2
+version: 5.0.3
```
Comment (tompizmor) [inline stable/wordpress/Chart.yaml]:
```suggestion
version: 5.0.4
```

**Review by tompizmor — APPROVED** [review]
/ok-to-test
/lgtm

**Pull request merged**